### ISSUE TITLE
lint rule - assigning `str` to `Iterable[str]` or `Sequence[str]`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3759]
https://pyrefly.org/en/docs/error-kinds/#string-as-iterable

this is very likely a mistake (and one that i make often), so it would be nice if ty had a lint rule for this (ideally this would be a ruff rule but it needs type information):

```py
from typing import Iterable

def takes_items(xs: Iterable[str]) -> None:
    ...

takes_items("hello")  # should be an error
```

**Comment by AlexWaygood** [comment]
Would probably need to be off by default, but yes, I agree that this would be really useful

**Comment by MichaReiser** [comment]
It could also be warning by default. Or are there common pattern where this is indeed what someone wanted?

**Comment by charliermarsh** [comment]
I also want this.

**Comment by correctmost** [comment]
> It could also be warning by default. Or are there common pattern where this is indeed what someone wanted?

I disabled Pyrefly's warning on code like this:

```python
import string
from collections import Counter

chars: Counter[str] = Counter()

chars.update(string.hexdigits)   # error: Passing `str` to `Iterable[str]` treats the string as an iterable of characters [string-as-iterable]
chars.update(string.whitespace)  # error: Passing `str` to `Iterable[str]` treats the string as an iterable of characters [string-as-iterable]
...
```

**Comment by MeGaGiGaGon** [comment]
I use this in my parser generators for both a literal sequence, and any of a few characters:
```py
from collections.abc import Sequence, Callable

class Parser[I, O, E]:
    def __init__(self, func: Callable[[I], O | E]):
        self._func = func
    def map_ok[NEW_O](self, func: Callable[[O], NEW_O]) -> Parser[I, NEW_O, E]:
        raise NotImplementedError

def just_seq[I](value: Sequence[I]) -> Parser[I, Sequence[I], None]:
    raise NotImplementedError
def any_of[I](values: Sequence[I]) -> Parser[I, I, None]:
    raise NotImplementedError

hello_parser: Parser[str, str, None] = just_seq("hello").map_ok("".join)
whitespace: Parser[str, str, None] = any_of(" \r\n\t")
```
In general it seems like this would be undesirable anytime you actually want a sequence of "characters".
(Pyrefly actually doesn't raise on this, I presume because it's generic)
